imports/timer/shared.lua

```diff
@@ -92,8 +93,13 @@ function timer:isPaused()
     return self.private.paused
 end
 
-function timer:restart(async)
-    self:forceEnd(false)
+function timer:restart(async, onEnd)
+    self:forceEnd(not not onEnd)
+
+    if type(onEnd) == 'function' then
+        self.private.onEnd = onEnd
+    end
```

Comment:
I don't think dynamically adding an onEnd function during a reset makes a lot of sense. What would this use case actually be?

@thelindat what are your thoughts on allowing an onEnd function to be added during a reset? Personally it feels like a really confusing user experience / flow.